email/src/main/java/edu/jhuapl/dorset/demos/EmailManager.java

```diff
@@ -0,0 +1,311 @@
+            logger.error("Failed to initialize folders");
+        }
+    }
+
+    /**
+     * Get folder
+     * 
+     * @param folder   the folder to be returned. Expressed by an integer value
+     * @return Folder   the folder to be accessed
+     */
+    private Folder getFolder(FolderType folder) {
+        switch (folder) {
+            case INBOX:
+                return inboxFolder;
+            case ERROR:
+                return errorFolder;
+            case COMPLETE:
+                return completeFolder;
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * Get number of messages in specified folder
+     * 
+     * @param folder   the folder to look in
+     * @return the number of messages left in folder
+     */
+    public int getCount(FolderType folder) {
+        try {
+            System.out.println(getFolder(folder));
+            return getFolder(folder).getMessageCount();
+        } catch (MessagingException exception) {
+            logger.error("Failed to count emails");
+            return -1;
+        }
+    }
+    
+    /**
+     * Get a message from a folder
+     * 
+     * @param folder   the folder to retrieve a message from
+     * @return the message
+     */
+    public Message getMessage(FolderType folder) {
+        try {
+            return getFolder(folder).getMessage(1);
+        } catch (MessagingException exception) {
+            logger.error("Failed to retrieve email");
+            return null;
+        }  
+    }
+    
+    /**
+     * Mark an email as seen
+     * 
+     * @param msg  the message to be marked seen
+     */
+    //not currently being used, but may be needed for threading
+    public void markSeen(Message msg) {
+        try {
+            msg.setFlag(Flags.Flag.SEEN, true);
+        } catch (MessagingException exception) {
+            logger.error("Failed to mark email as seen");
+        }
+    }
+
+    /**
+     * Retrieve and return text from an email
+     * 
+     * @param msg   the message to be read
+     * @return msgContent  the text the email
+     */
+    public String readEmail(Message msg) {
+        try {
+            System.out.println("---------------------------------");
+            System.out.println("From: " + msg.getFrom()[0]);
+            System.out.println("Subject: " + msg.getSubject());
+    
+            String msgContent = "";
+            response = "";
+            msgContent += writePart(msg);
+            System.out.println("Email reads: " + msgContent);
+            return msgContent;
+        } catch (MessagingException exception) {
+            logger.error("Failed to fetch email body");
+            return "Error occured while reading email";
+        }
+    }
+
+    /**
+     * Retrieve the body of an email
+     * 
+     * @param part   the email body to be retrieved
+     * @return response   the text of an email
+     */
+    private String writePart(Part part) {
```

Comment:
This method name makes me think the code is wiring data to a message rather than reading it